### PR TITLE
[#1855] Restore coinholder voting on zcash_voting 2.0 with Ironwood support

### PR DESCRIPTION
**Pull request by LukasKorba**
Restores coinholder voting (#1855) on `zcash_voting = "=2.0.0-rc.5"` with Ironwood support, consumed by the Zodl app's `chp-re-enable` branch (MOB-1678).

## What this is

- **rc.3 → rc.5 pin** with the crate-2.0 adaptations (`connect_pir_blocking` fail-closed PIR handshake, `PirLayout` threading).
- **De-mirrored wire layer**: the SDK stops re-typing the crate's wire structs; submission/recovery JSON passes through crate-owned `to_wire_json` / `recover_wire_json` (`rust/src/voting/`, `Sources/ZcashLightClientKit/Rust/Voting/`).
- **Confirm/recover passthroughs** (`confirm_vote_submission`, `recover_wire_json`) and the **software delegation-signing FFI** `zcashlc_voting_sign_delegation_request` (seed borrowed, never copied or logged).
- **Recoverable-share-indices FFI** `zcashlc_voting_recoverable_share_indices`: recovery consumers enumerate the crate's actual `enc_share` list (single-share slice respected) instead of guessing share counts from round parameters.
- **Ironwood tree-root fix** (`aec99172`): a voting round's `nc_root` is the Ironwood note-commitment-tree root; snapshot validation goes through the crate's `witness::generate_note_witnesses`, which reads the Ironwood tree. ⚠ Note for maintainers: this re-lands what `8a40d1f9` (Jul 17) fixed and the merge `eea6cde8` (Aug 6) silently reverted — **`main` carries that regression today**; this branch restores the fix plus the deleted regression tests, re-seeded on testnet/NU6.3 fixtures with a decoy Orchard tree.

## Evidence

- `cargo test --lib voting`: 84/84 (incl. 10 negative-path FFI tests from review round 1). `swift test --filter OfflineTests`: 873/0 (both re-measured on this head, 2026-08-13). FFI symbol surface additive only vs. pre-branch (+`zcashlc_voting_recoverable_share_indices`); review-round commit is symbol-identical (nm diff empty).
- Live: first complete end-to-end software vote on Valar stage (delegation + 2 cast votes confirmed on-chain + 32/32 tally shares accepted), 2026-08-12, via the Zodl branch that consumes this SDK.
- **2026-08-14: rebased onto `main` @ `785c7618`** — main merged #1971 (the escalated broadcast-readback fix) this morning; all 8 patches byte-equal by range-diff, sole conflict CHANGELOG (auto-merged, verified single `Unreleased` heading with both sides' entries). `cargo test --lib voting` re-run on this head: 84/84. Integrated-stack gates on the stacked #1972 head: OfflineTests 891/0 (main's 17 new broadcaster/readback tests included), zodl suite 1301/188.

## Where to look

- `rust/src/voting/` — the FFI voting layer (delegation.rs, util.rs, rounds.rs, helpers.rs).
- `Sources/ZcashLightClientKit/Rust/Voting/` — Swift wrappers + DTOs.
- The Zodl-side campaign record documents every finding with commit refs; it lives out-of-tree per the `.plans/` convention (available on request).

---

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [ ] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [ ] Run the app: Did you run the app and try the changes?
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [ ] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/mater/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage befor and after your changes and when possible, leave it in a better place. [Learn More...](../blob/master/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/master/README.md), [LICENSE.md](../blob/master/LICENSE.md), and [Architecture.md](../blob/master/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift [hunk 1]
```diff
@@ -295,13 +307,19 @@ public struct VotingDelegationSubmission: Codable, Sendable {
...
 
-/// Everything produced by committing one cast vote: the signed commitment
-/// fields destined for the vote chain, the encrypted shares the vote proof
-/// binds, and the helper-server payloads derived from them.
+/// The result of committing one cast vote: the signed commitment fields destined
+/// for the vote chain, and the encrypted shares the vote proof binds.
+///
+/// Helper-server payloads are deliberately not here. A commit made before the
+/// vote's tree position is confirmed can only produce provisional payloads, and
+/// sending those is the bug the sequence in `CHP_DESIGN.md` §3/A2 exists to
```
Comment (dannywillems) [inline Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift]: This is part of AI generated plan; I would suggest to remove this information as it cannot be tracked anywhere in the repository. It is some kinds of dead information.

File: Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift [hunk 2]
```diff
@@ -267,26 +267,38 @@ public struct VotingDelegationPirPrecomputeResult: Codable, Sendable {
...
+///
+/// The FFI returns `zcash_voting::wire::DelegationSubmissionWire` serialized by
+/// the crate, so the field names and the base64 encoding are the crate's and the
+/// SDK reshapes nothing. Two consequences for callers: `tx1Effects` — the
+/// versioned Ironwood TX1 effecting data the vote chain requires, and whose
+/// absence is the `400: tx1 effects must be 821 bytes, got 0` rejection — is
+/// present without anyone assembling it, and the legacy `sighash` field is gone
+/// from the wire. The signer's sighash still exists; it simply never belonged in
+/// the submission body, because the server derives the signing digest itself.
 public struct VotingDelegationSubmission: Codable, Sendable {
```
Comment (dannywillems) [inline Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift:285]: Nit: as stated in the doc, this is exactly https://github.com/valargroup/zcash_voting/blob/main/zcash_voting/src/wire.rs#L72, with different variable name. It would be nice to keep the same variable name.

File: Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift [hunk 3]
```diff
@@ -267,26 +267,38 @@ public struct VotingDelegationPirPrecomputeResult: Codable, Sendable {
...
+    /// Versioned Ironwood TX1 effecting data, base64 (821 bytes decoded).
+    public let tx1Effects: String
+    public let nfSigned: String
+    public let cmxNew: String
+    public let govComm: String
+    public let govNullifiers: [String]
+    public let proof: String
     public let voteRoundId: String
 
     enum CodingKeys: String, CodingKey {
```
Comment (dannywillems) [inline Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift:299]: Reviewer note: this is to match the renaming [here](https://github.com/valargroup/zcash_voting/blob/main/zcash_voting/src/wire.rs#L72-L85).

File: Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift [hunk 4]
```diff
@@ -267,26 +267,38 @@ public struct VotingDelegationPirPrecomputeResult: Codable, Sendable {
...
         case spendAuthSig = "spend_auth_sig"
-        case sighash
-        case nfSigned = "nf_signed"
+        case tx1Effects = "tx1_effects"
+        case nfSigned = "signed_note_nullifier"
         case cmxNew = "cmx_new"
-        case govComm = "gov_comm"
+        case govComm = "van_cmx"
         case govNullifiers = "gov_nullifiers"
         case proof
```
Comment (dannywillems) [inline Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift:307]: Reviewer note: I guess when there is no string matching, it means that it matches exactly as the variable name.

File: Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift [hunk 5]
```diff
@@ -295,13 +307,19 @@ public struct VotingDelegationSubmission: Codable, Sendable {
...
+/// ``VotingRustBackend/recoverWireJson(commitmentBundleJson:proposalId:shareIndex:voteCommitmentTreePosition:submitAt:)``
+/// after ``VotingRustBackend/confirmVoteSubmission(roundId:bundleIndex:proposalId:txHash:eventsJson:)``.
 ///
-/// Every field here is wire data — it is published on chain or sent to helper
-/// servers — so the commit result carries no secret the wallet must retain. The
-/// signing secrets used to produce it stay inside `zcash_voting`.
+/// Every field here is wire data — it is published on chain — so the commit
+/// result carries no secret the wallet must retain. The signing secrets used to
+/// produce it stay inside `zcash_voting`.
 public struct VotingVoteCommit: Codable, Sendable {
```
Comment (dannywillems) [inline Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift:332]: This is to match https://github.com/valargroup/zcash_voting/blob/main/zcash_voting/src/wire.rs#L88-L99. It would be nice to have the doc mentioning it, and have the same naming convention.

File: Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift
Comment (dannywillems) [inline Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift:350]: If I'm not mistaken, and if I understand correctly the mapping, this is not the right mapping, see https://github.com/valargroup/zcash_voting/blob/main/zcash_voting/src/wire.rs#L95.

File: Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift [hunk 6]
```diff
@@ -295,13 +307,19 @@ public struct VotingDelegationSubmission: Codable, Sendable {
...
     public let proposalId: UInt32
```
Comment (dannywillems) [inline Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift:333]: Nit: in the original structure, proposal_id comes after `vote_commitment`

File: Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift (hunk 5 above)
Comment (dannywillems) [reply Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift:332]: Ah, it seems I'm wrong! Ignore my comments below.

File: Tests/OfflineTests/VotingConfirmVoteSubmissionTests.swift [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+private let confirmWalletId = "test-wallet"
+private let confirmNetworkId: UInt32 = 1
+/// A well-formed round identifier that is never initialized, so the crate has no
+/// vote row to confirm against.
+private let confirmMissingRoundId = hexRoundId(0xfd)
+private let confirmTxHash = "vote-tx-hash"
+/// One well-formed `cast_vote` event, shaped exactly as
+/// `Vec<zcash_voting::confirmation::TxEvent>` deserializes it. The FFI must hand
+/// it to the crate unparsed — the SDK never splits `leaf_index` itself.
+private let confirmEventsJson = """
```
Comment (dannywillems) [inline Tests/OfflineTests/VotingConfirmVoteSubmissionTests.swift:25]: Nit: JSON can be generated by scripts from the Rust code before executing the tests, with expected output. These can be used as evolving test vectors. This could be done for all wire types.

File: Cargo.lock [hunk 8]
```diff
@@ -30,10 +30,10 @@ version = "0.8.4"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "b169f7a6d4742236a0a00c541b845991d0ac43e546831af1249753ab4c3aa3a0"
 dependencies = [
-"cfg-if",
```
Comment (dannywillems) [inline Cargo.lock:33]: I'm surprised about this diff. The one space is the expected behavior from Cargo.lock.
Has the file been manually modified in a previous patch?

File: Cargo.lock [hunk 9]
```diff
@@ -2154,12 +2154,12 @@ version = "0.28.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "51255ea7cfaadb6c5f1528d43e92a82acb2b96c43365989a28b2d44ee38f8795"
 dependencies = [
-"ash",
-"hashbrown 0.16.1",
-"log",
-"presser",
-"thiserror 2.0.19",
-"windows 0.61.3",
```
Comment (dannywillems) [inline Cargo.lock:2162]: Why this diff?

File: Cargo.lock [hunk 10]
```diff
@@ -8494,3 +8440,8 @@ dependencies = [
  "cc",
  "pkg-config",
 ]
+
+[[patch.unused]]
```
Comment (dannywillems) [inline Cargo.lock:8444]: Expected in this diff?

File: rust/src/voting/delegation.rs [hunk 11]
```diff
@@ -539,7 +470,12 @@ pub unsafe extern "C" fn zcashlc_voting_build_and_prove_delegation(
         })?;
         let round_name_str = unsafe { str_from_ptr(round_name, round_name_len) }?;
         let pir_url = unsafe { str_from_ptr(pir_server_url, pir_server_url_len) }?;
-        let pir_client = connect_pir_client(&pir_url)?;
+        let pir_layout = voting::config::PirLayout {
```
Comment (dannywillems) [inline rust/src/voting/delegation.rs:473]: Reviewer note: it seems that the zcash_voting SDK exposes all the fields publicly. It differs from librustzcash and other Zcash libraries, see https://github.com/valargroup/zcash_voting/blob/main/zcash_voting/src/config/mod.rs#L129-L135.

**Review by dannywillems — DISMISSED** [review]
utACK 6f6018b2eb56a3a33d613f8d258cf051c31fc2dc. Approving to move forward. Left a few non-blocking comments for potential future iterations. The app equivalent is on https://github.com/zodl-inc/zodl-ios/pull/1988. I haven't tested.

File: rust/src/voting/util.rs [hunk 12]
```diff
@@ -582,6 +588,72 @@ mod tests {
...
     }
 
+    /// Voting rounds are anchored to the **Ironwood** note commitment tree, so
+    /// when the cached `TreeState` carries both pools the extracted `nc_root`
+    /// must be the Ironwood root, not the Orchard one. This is the second half
+    /// of the `8a40d1f9` fix that the `eea6cde8` merge lost; without it, nothing
+    /// in the suite notices which field this FFI reads.
+    #[test]
+    fn extract_nc_root_returns_ironwood_root_when_both_trees_present() {
+        use ff::PrimeField;
```
Comment (dannywillems) [inline rust/src/voting/util.rs]: Nit: move this up. Don't keep it in function body.

File: rust/src/voting/util.rs [hunk 13]
```diff
@@ -582,6 +588,72 @@ mod tests {
...
+        use incrementalmerkletree::frontier::{CommitmentTree, Frontier};
+        use orchard::tree::MerkleHashOrchard;
+        use pasta_curves::pallas;
+        use zcash_primitives::merkle_tree::write_commitment_tree;
+
+        const TREE_DEPTH: u8 = orchard::NOTE_COMMITMENT_TREE_DEPTH as u8;
+
+        fn merkle_hash(tag: u64) -> MerkleHashOrchard {
```
Comment (dannywillems) [inline rust/src/voting/util.rs]: Nit: move this up. Never keep methods defined in function body.

File: rust/src/voting/util.rs [hunk 14]
```diff
@@ -582,6 +588,72 @@ mod tests {
...
+            let repr = pallas::Base::from(tag).to_repr();
+            MerkleHashOrchard::from_bytes(&repr).expect("small field element is canonical")
+        }
+
+        fn frontier_with(tags: &[u64]) -> Frontier<MerkleHashOrchard, TREE_DEPTH> {
```
Comment (dannywillems) [inline rust/src/voting/util.rs]: Nit: move this up. Never keep methods defined in function body.

File: rust/src/voting/util.rs [hunk 15]
```diff
@@ -582,6 +588,72 @@ mod tests {
...
+            let mut frontier = Frontier::empty();
+            for tag in tags {
+                assert!(frontier.append(merkle_hash(*tag)));
+            }
+            frontier
+        }
+
+        fn tree_hex(frontier: &Frontier<MerkleHashOrchard, TREE_DEPTH>) -> String {
```
Comment (dannywillems) [inline rust/src/voting/util.rs]: Nit: move this up. Never keep methods defined in function body.

File: rust/src/voting/signing.rs [hunk 16]
```diff
@@ -0,0 +1,177 @@
...
+///
+/// - `db` must be a valid, non-null `VotingDatabaseHandle` pointer.
+/// - For every `(ptr, len)` byte argument (`round_id`, `fvk_bytes`,
+///   `hotkey_stored_secret`, `seed_fingerprint`, `round_name`, `seed`): if
+///   `len > 0` then `ptr` must be non-null and valid for reads for `len` bytes;
+///   if `len == 0`, `ptr` is ignored.
+/// - `seed` must remain valid and unmutated for the duration of the call. The
+///   callee neither retains nor frees it.
+#[unsafe(no_mangle)]
+pub unsafe extern "C" fn zcashlc_voting_sign_delegation_request(
```
Comment (dannywillems) [inline rust/src/voting/signing.rs:80]: Nit: for all FFI methods, it would be nice to actually tests all edge cases where we pass wrong values, and check if the expected errors are raised.

File: Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift (hunk 1 above)
Comment (LukasKorba) [reply Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift]: Right — that referenced campaign planning material that never shipped with the repo. Reworded in 42751e54 so the warning stands alone: provisional payloads made before the vote's tree position is confirmed must never be sent to a helper server.

File: Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift (hunk 2 above)
Comment (LukasKorba) [reply Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift:285]: Doc half addressed in 42751e54: the header now states it mirrors `zcash_voting::wire::DelegationSubmissionWire` and that the CodingKeys carry the crate's serde field names where Swift differs (`signed_note_nullifier`, `van_cmx`). The renames themselves are deferred deliberately: these are public members the app consumes, so that's a breaking-API wave with a MIGRATING.md entry — natural slot is the next crate bump.

File: Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift (hunk 5 above)
Comment (LukasKorba) [reply Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift:332]: Doc added in 42751e54, with one precision your pointer surfaced: `VotingVoteCommit` does not mirror `wire::VoteCommitmentWire` — that type is base64-string encoded, serde-renames `anchor_height` to `vote_comm_tree_anchor_height`, carries `vote_round_id`, and has no `enc_shares`. What this struct decodes is the FFI's `JsonVoteCommit` (rust/src/voting/json.rs), the JSON shape of `zcash_voting::vote::VoteCommit` minus its provisional `share_payloads`. The doc now says exactly that.

File: Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift
Comment (LukasKorba) [reply Sources/ZcashLightClientKit/Rust/Voting/VotingTypes.swift:350]: Verified anyway to be safe: `voteKeyRandomizer = "r_vpk"` matches the crate's own field name (`vote::VoteCommit.r_vpk`, same spelling as `wire::VoteCommitmentWire.r_vpk` at wire.rs:96), and member order can't matter — serde_json and JSONDecoder both resolve object members by key, never by position.

File: Cargo.lock (hunk 8 above)
Comment (LukasKorba) [reply Cargo.lock:33]: The direction is inverted: the missing leading space is on `main`'s copy (a previous patch de-formatted it), and this branch's relock restored cargo's canonical format. So yes — manually modified previously, on main; this diff is cargo healing it.

File: Cargo.lock (hunk 9 above)
Comment (LukasKorba) [reply Cargo.lock:2162]: This one is a `windows` re-resolution under `gpu-allocator` (0.61.3 → 0.62.2 — both versions already present in both locks; cargo re-picked during the relock). The lock's other real changes are the rc.3→rc.5 `zcash_voting` family bumps: `pir-client`/`pir-types` → 0.4.0-rc.4, `vote-commitment-tree(-client)` → 0.4.0-rc.2/0.6.0-rc.2, `imt-tree` → 0.2.1. Nothing unexplained.

File: Cargo.lock (hunk 10 above)
Comment (LukasKorba) [reply Cargo.lock:8444]: Expected, and informative: the `[patch.crates-io] slipstream-core → slipstream-internal` stanza is main's (`0076d56a`); it matches no crates-io package in this graph (the real dep is `zodl-slipstream` renamed to `slipstream-core`), so the relock recorded it as `[[patch.unused]]` and pruned two orphaned lock entries main carried with zero dependents (`slipstream-core 0.6.4`, `slipstream-gpuhash`). Removing the stanza itself belongs in a main-targeted cleanup — `origin/fix-cargo-lock` looks to be tending that lock already.

File: Tests/OfflineTests/VotingConfirmVoteSubmissionTests.swift (hunk 7 above)
Comment (LukasKorba) [reply Tests/OfflineTests/VotingConfirmVoteSubmissionTests.swift:25]: Good pattern — noted as a follow-up iteration for all wire types. It wants a small Rust harness emitting vectors at test time, which deserves its own PR rather than riding this one.

File: rust/src/voting/util.rs (hunk 12 above)
Comment (LukasKorba) [reply rust/src/voting/util.rs]: Hoisted in 42751e54 — the three nested helpers (`merkle_hash`, `frontier_with`, `tree_hex`) plus their `TREE_DEPTH` const and function-local imports now sit at test-module scope above the test. (This flagged line was one of the `use` statements; it moved with them.)

File: rust/src/voting/util.rs (hunk 13 above)
Comment (LukasKorba) [reply rust/src/voting/util.rs]: Hoisted in 42751e54 (details in the thread at the first occurrence).

File: rust/src/voting/util.rs (hunk 14 above)
Comment (LukasKorba) [reply rust/src/voting/util.rs]: Hoisted in 42751e54.

File: rust/src/voting/util.rs (hunk 15 above)
Comment (LukasKorba) [reply rust/src/voting/util.rs]: Hoisted in 42751e54.

File: rust/src/voting/signing.rs (hunk 16 above)
Comment (LukasKorba) [reply rust/src/voting/signing.rs:80]: Started bounded in 42751e54: ten negative-path tests for `zcashlc_voting_sign_delegation_request` — null handle, invalid-UTF-8 round id, wrong-length seed fingerprint, wrong-sized hotkey secret, unknown round, bundle without stored signing data, mismatched wallet-seed fingerprint, non-ZIP-32 seed, non-canonical stored alpha — plus a positive control proving the planted fixtures reach the deep checks (so those tests can't pass vacuously). The sweep across the remaining FFI files is a follow-up iteration.

File: rust/src/voting/delegation.rs (hunk 11 above)
Comment (LukasKorba) [reply rust/src/voting/delegation.rs:473]: Agreed it diverges from librustzcash norms — that's `zcash_voting`'s own design (exact-pinned here, never modified in this repo). Worth raising with the crate's maintainers; nothing the SDK layer can change locally.

File: Tests/OfflineTests/VotingConfirmVoteSubmissionTests.swift (hunk 7 above)
Comment (dannywillems) [reply Tests/OfflineTests/VotingConfirmVoteSubmissionTests.swift:25]: See https://github.com/zcash/zcash-swift-wallet-sdk/issues/1963

**Review by dannywillems — DISMISSED** [review]
utACK 0798292543b858ee68b4637dc1e97686b5260471

**Review by dannywillems — APPROVED** [review]
utACK d3716737a3861d114db9c27449c6084a1d94f189
